### PR TITLE
uplink ringtone code + minor cleanup

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
removed the tator-only toggle uplink button from pda settings
tators are told their ringtone code for uplink
they can enter it or guess the 625 combinations to open uplink ui (no #s)

also did some very small cleanup of related files imports

**Media**

https://user-images.githubusercontent.com/39013340/230923187-66476a7d-1979-45f0-8211-1ad6ed5cde52.mp4

- [X] I have added a video to this PR showcasing its changes ingame

**Changelog**
:cl:
- add: The Syndicate has added a ringtone code lock to uplinks. Operatives are told the code when they are briefed. Set your ringtone to the code to open your uplink!

### CODE REVIEW

**Comment by AJCM-git** [comment]
The way this should probably work is by adding/removing the uplink button from the PDA BUI imo (also some UX QOL stuff might be needed now that the code is very important, like adding it to the character menu, automatically opening the character menu when you are selected as traitor and coloring codewords/code in red or smt in the chat)

**Comment by deltanedas** [comment]
locking/unlocking would be cool however
1. the old bui code for uplink is horrible

but ill try

**Comment by deltanedas** [comment]
other stuff:

>  automatically opening the character menu when you are selected as traitor
good thing to have and sounds easy, i will try to do it

> coloring codewords/code in red or smt in the chat

im split on this:
- for good players this makes it incredibly easy to find traitors since you dont need to always remember to check for codewords
- but i shouldnt need to deeply analyze chat 24/7 as a traitor, leave that for bozos that realise a tator tried a codeword on them, they dont have the special brain implants or whatever

**Comment by deltanedas** [comment]
@AJCM-git so uh is there an existing thing for telling the client to open a window or should i make an event specifically for opening character info then handle it on client

**Comment by AJCM-git** [comment]
> other stuff:
> 
> > automatically opening the character menu when you are selected as traitor
> 
> good thing to have and sounds easy, i will try to do it
> 
> > coloring codewords/code in red or smt in the chat
> 
> im split on this:
> 
>     * for good players this makes it incredibly easy to find traitors since you dont need to always remember to check for codewords
> 
>     * but i shouldnt need to deeply analyze chat 24/7 as a traitor, leave that for bozos that realise a tator tried a codeword on them, they dont have the special brain implants or whatever
> 
>     * at that point you may as well put a traitor icon next to your sprite so it saves you having to say a highlighted codeword in bar

i meant highlighting the ones that appear in the chat the first time informing you what they are, but if you manage to highlight the ones in the chat that would be a cool bonus


> @AJCM-git so uh is there an existing thing for telling the client to open a window or should i make an event specifically for opening character info then handle it on client

I think UISystem has a method for it

File: Content.Server/PDA/Ringer/RingerSystem.cs [hunk 1]
```diff
@@ -46,7 +53,31 @@ private void OnSetRingtone(EntityUid uid, RingerComponent ringer, RingerSetRingt
...
         }
 
+        private void OnSetUplinkRingtone(EntityUid uid, RingerUplinkComponent uplink, RingerSetRingtoneMessage args)
+        {
+            if (uplink.Code.SequenceEqual(args.Ringtone) &&
+                args.Session.AttachedEntity != null &&
+                TryComp<StoreComponent>(uid, out var store))
+            {
+                var user = args.Session.AttachedEntity.Value;
+                Logger.Info($"{ToPrettyString(user):player} opened uplink {ToPrettyString(uid):uplink}");
```
Comment (mirrorcult) [inline Content.Server/PDA/Ringer/RingerSystem.cs:63]: no need loggy

**Review by mirrorcult — APPROVED** [review]

**Comment by deltanedas** [comment]
@mirrorcult so should i change it to be lock rather than open or is it ok

**Comment by mirrorcult** [comment]
the way it works now is fine

**Comment by crazybrain23** [comment]
Does this mean you can't steal TC from another syndicates PDA without knowing the code?

**Comment by Emisse** [comment]
might be a good time to start thinking of implementing customized uplinks like ss13; implant with less tc, pen uplink, etc

**Comment by deltanedas** [comment]
> Does this mean you can't steal TC from another syndicates PDA without knowing the code?

correct. you either need to:
- brute force it like a caveman
- brute force it with macro and get bwoinked
- hope bozo left code as ringtone
- get bozo to tell you the code somehow
this is much nicer than previously just rob ssds pda then check uplink + loot the 20tc

> might be a good time to start thinking of implementing customized uplinks like ss13; implant with less tc, pen uplink, etc

i dont think these would have any code in common with pda uplink stuff:
- implant likely an action that toggles ui on your entity
- pen likely toggles ui on itself when clicked (should probably change to writing password on paper thats pretty easily checked)
- headset (since frequency isnt real i guess whispering passwoed into it)

only code theyd share here is trycomp actor and store toggleui

**Comment by keronshb** [comment]
This will come in handy when the mind refactor is in place so we can store codes there.

File: Resources/Locale/en-US/game-ticking/game-presets/preset-traitor.ftl [hunk 2]
```diff
@@ -48,6 +48,15 @@ traitor-role-greeting =
     Death to Nanotrasen!
 traitor-role-codewords =
     The codewords are:
-    {$codewords}.
+    [color=red]{$codewords}[/color].
```
Comment (keronshb) [inline Resources/Locale/en-US/game-ticking/game-presets/preset-traitor.ftl:51]: You can't change color like this here

**Review by keronshb — CHANGES_REQUESTED** [review]
image

There's a bug when you're making someone a traitor from the context menu. The codewords aren't appearing and it shows the attempted color changing code.

**Comment by deltanedas** [comment]
oh boy i wonder what broke it now

File: Resources/Locale/en-US/game-ticking/game-presets/preset-traitor.ftl (hunk 2 above)
Comment (deltanedas) [reply Resources/Locale/en-US/game-ticking/game-presets/preset-traitor.ftl:51]: is that not how you do it

Comment (keronshb) [reply Resources/Locale/en-US/game-ticking/game-presets/preset-traitor.ftl:51]: I don't think you can do it in general.

Comment (deltanedas) [reply Resources/Locale/en-US/game-ticking/game-presets/preset-traitor.ftl:51]: guess ill remove

**Comment by deltanedas** [comment]
i dont know why but removing color tag also fixed codewords

**Review by keronshb — APPROVED** [review]
Tested and works fine now.

**Comment by GoodWheatley** [comment]
I’d normally just ask this on Discord but you don’t have an account, so this seems like the next best method. Is everyone able to use an unlocked uplink, or are traitors still the only ones who can see the button?

**Comment by mirrorcult** [comment]
read the PR description, its not traitor-only anymore

**Pull request merged**